### ISSUE TITLE
Bumping Kafka to 3.2.0

### ISSUE THREAD

**Issue by jarohen** [xtdb/crux#2118]
Could try out KRaft, too

**Comment by jsulmont** [comment]
Hi @jarohen happy to take that one.
Just to mention it,  the versions of the Kafka Client lib and the Kafka broker are two different things.
Kafka Clients talk to the broker via a [wire protocol](https://kafka.apache.org/protocol.html). So what really matters, is if the version of the client lib you're using, talks the same protocol as the broker you're talking to.
Thankfully the protocol is fairly stable and doesn't change much (see [here](https://cwiki.apache.org/confluence/display/KAFKA/Compatibility+Matrix)).
TL;DR one could play a KRaft broker with Kafka client 3.1  (or earlier).
Fun fact, we could also try out RedPanda.

**Comment by jarohen** [comment]
Hey @jsulmont, feel free :slightly_smiling_face: 

Important thing for us is that we can work with whatever our users have installed, so we've tended to err on the side of earlier Kafka versions and rely on their backwards compatibility, but with Core2 we get the option of a fresh start. :slightly_smiling_face: Regarding RedPanda: from XT's PoV we don't really want to shoehorn users into one particular Kafka solution, so _depending_ on it is likely the wrong move for us, but no harm in seeing how compatible it is and having it as an option. 

Cheers!

James

**Comment by jarohen** [comment]
Fixed with xtdb/core2#656, cheers @jsulmont - we didn't need to upgrade our version in the end.
